cmd/dep/root_analyzer.go

```diff
@@ -53,6 +57,71 @@ func (a *rootAnalyzer) InitializeRootManifestAndLock(dir string, pr gps.ProjectR
 	return
 }
 
+func (a *rootAnalyzer) cacheDeps(pr gps.ProjectRoot, pkgT pkgtree.PackageTree) {
+	packages := make(map[string]bool)
+	dependencies := make(map[gps.ProjectRoot][]string)
+	logger := a.ctx.Err
+
+	syncDep := func(pr gps.ProjectRoot, sm gps.SourceManager) {
+		message := fmt.Sprintf("Cached %s", pr)
```

Comment:
Instead of using fmt directly here and elsewhere in this file, use `a.ctx.Err`. This let's us capture/discard output more easily in our tests.